src/Features/Core/Portable/ReplaceDocCommentTextWithTag/AbstractReplaceDocCommentTextWithTagCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,251 @@
+            CodeRefactoringContext context, SemanticModel semanticModel, SyntaxToken token, TextSpan replacementSpan, string text)
+        {
+            var parsed = ParseExpression(text);
+            var foundSymbol = semanticModel.GetSpeculativeSymbolInfo(token.SpanStart, parsed, SpeculativeBindingOption.BindAsExpression).GetAnySymbol();
+            if (foundSymbol == null)
+            {
+                return false;
+            }
+
+            RegisterRefactoring(context, replacementSpan, $@"<see cref=""{text}""/>");
+            return true;
+        }
+
+        private ISymbol GetEnclosingSymbol(SemanticModel semanticModel, int position, CancellationToken cancellationToken)
+        {
+            var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
+            var token = root.FindToken(position);
+
+            for (var node = token.Parent; node != null; node = node.Parent)
+            {
+                if (semanticModel.GetDeclaredSymbol(node) is ISymbol declaration)
+                {
+                    return declaration;
+                }
+            }
+
+            return null;
+        }
+
+        private void RegisterRefactoring(
+            CodeRefactoringContext context, TextSpan expandedSpan, string replacement)
+        {
+            context.RegisterRefactoring(new MyCodeAction(
+                string.Format(FeaturesResources.Use_0, replacement),
+                c => ReplaceTextAsync(context.Document, expandedSpan, replacement, c)));
+        }
+
+        //private bool TryMatchTextAsSymbol(
+        //    CodeRefactoringContext context, TextSpan expandedSpan, 
+        //    string text, SemanticModel semanticModel, 
+        //    int speculativePosition, SpeculativeBindingOption binding)
+        //{
+        //    var parsed = ParseExpression(text);
+
+        //    var symbolInfo = semanticModel.GetSpeculativeSymbolInfo(speculativePosition, parsed, binding);
+        //    switch (symbolInfo.GetAnySymbol())
+        //    {
+        //        case IParameterSymbol parameter:
+                    
+        //            return true;
+        //        case ITypeParameterSymbol typeParameter:
+        //            return true;
+        //        case ISymbol symbol:
+        //            RegisterRefactoring(context, expandedSpan, $@"<see cref=""{text}""/>");
+        //            return true;
+        //        default:
+        //            return false;
+        //    }
+        //}
+
+        private async Task<Document> ReplaceTextAsync(
+            Document document, TextSpan span, string replacement, CancellationToken cancellationToken)
+        {
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var newText = text.Replace(span, replacement);
+
+            return document.WithText(newText);
+        }
+
+        private TextSpan ExpandSpan(SourceText sourceText, TextSpan span, bool fullyQualifiedName)
```

Comment:
I just want to point that this is not surrogate pairs friendly. Might need fixing with #13560.